src/main/java/com/microsoft/jenkins/containeragents/aci/AciCloud.java

```diff
@@ -236,6 +246,22 @@ public String getCredentialsId() {
         return credentialsId;
     }
 
+    public String getLogAnalyticsCredentialsId() {
+        return logAnalyticsCredentialsId;
+      }
+
+    public StandardUsernamePasswordCredentials getLogAnalyticsCredentials() {
```

Comment:
normally you would have the credential ID in the model class and then the service would load it, this may interfere with the configuration as code plugin if left here.

I would suggest moving it to AciService and extracting this code out to a method
https://github.com/jenkinsci/azure-container-agents-plugin/blob/920e4546d0be9eb8a14775847b8bde01bbc74365/src/main/java/com/microsoft/jenkins/containeragents/aci/AciService.java#L187